packages/api/core/src/api/init.ts

```diff
@@ -118,17 +120,32 @@ export default async ({ dir = process.cwd(), interactive = false, copyCIFiles =
                   }
                   await installDepList(dir, templateModule.devDependencies || [], DepType.DEV);
                 },
+                exitOnError: false,
               },
               {
                 title: 'Finalizing dependencies',
                 task: async (_, task) => {
-                  await initNPM(dir, task);
+                  return task.newListr([
+                    {
+                      title: 'Installing common dependencies',
+                      task: async (_, task) => {
+                        await initNPM(dir, task);
+                      },
+                      exitOnError: false,
+                    },
+                    {
+                      title: process.env.LINK_FORGE_DEPENDENCIES_ON_INIT ? 'Linking forge dependencies' : 'Skip linking forge dependencies',
+                      task: async (_, task) => {
+                        await initLink(dir, task);
+                      },
+                      exitOnError: true,
```

Comment:
Separated the linking task, and made it `exitOnError: true` so it no longer silently fails in tests. It will only run when `LINK_FORGE_DEPENDENCIES_ON_INIT` is truthy.